odoo/modules/db.py

```diff
@@ -138,14 +138,29 @@ def create_categories(cr, categories):
     return p_id
 
 def has_unaccent(cr):
-    """ Test if the database has an unaccent function.
+    """ Test if the database has an unaccent function and returning it `provolatile` if it exists.
 
     The unaccent is supposed to be provided by the PostgreSQL unaccent contrib
     module but any similar function will be picked by OpenERP.
 
+    The `provolatile` of unaccent allows to know if the unaccent function
+    can be used to create index (it should be 'i' - means immutable), see
+    https://www.postgresql.org/docs/current/catalog-pg-proc.html.
+
+    :return: False if unaccent doesn't exist, `provolatile` of the unaccent method otherwise (always truly)
```

Comment:
Returning pg catalog detail values is super gnarly.

Maybe an enum with "No" (falsy), "Volatile", "Immutable", or something along those lines? Missing/Present/Indexable to better match the semantics we need? And fix the doc to talk about things in those terms?